server/request_handler.go

```diff
@@ -21,36 +20,38 @@ type RpcRequestHandler struct {
 	timeStarted     time.Time
 	defaultProxyUrl string
 	relaySigningKey *ecdsa.PrivateKey
-	uid             string
+	uid             uuid.UUID
+	db              database.Store
+	requestRecord   *requestRecord
 }
 
-func NewRpcRequestHandler(respw *http.ResponseWriter, req *http.Request, proxyUrl string, relaySigningKey *ecdsa.PrivateKey) *RpcRequestHandler {
+func NewRpcRequestHandler(respw *http.ResponseWriter, req *http.Request, proxyUrl string, relaySigningKey *ecdsa.PrivateKey, db database.Store, requestRecord *requestRecord) *RpcRequestHandler {
```

Comment:
requestRecord should be created here, not passed in from the outside?

Reply:
updated